### DISCUSSION TITLE
How to point ArgoCD to a specific commit in gitops-engine?

### DISCUSSION THREAD

**Discussion by andrii-korotkov-verkada** [argoproj/argo-cd#18971] [Q&A] [answered]
I'm working on a few gitops-engine PRs (https://github.com/argoproj/gitops-engine/pull/601, https://github.com/argoproj/gitops-engine/pull/603) and would like to use them in ArgoCD once merged. 
I see references to gitops-engine in `go.sum` and `go.mod` like: `v0.7.1-0.20240628155502-fa0e8d60a3a4`. I have a few questions:
* Why is v0.7.1 used and not the latest v0.7.3?
* Is the 2nd part based on the date and time of the commit? Is it in UTC?
* Is the 3rd part a part of a commit hash?
* Do I have to do anything special to make a commit to be possible to be referenced like this?
Thank you.

**Comment by crenshaw-dev** [comment, marked as answer]
> Why is v0.7.1 used and not the latest v0.7.3?

We abandoned actual versions a while ago. We just point to specific commits now. 

> Is the 2nd part based on the date and time of the commit? Is it in UTC?

That's my understanding, but I'm not sure. It's generated by go tooling. 

> Is the 3rd part a part of a commit hash?

Yep!

> Do I have to do anything special to make a commit to be possible to be referenced like this?

Nope. If the commit is on your fork, use a `replace` in go.mod to point to your fork. Replace the version/time/hash bit with the full hash from your fork. Then run `go mod tidy`. The go command will go find your commit and reformat go.mod to have the version/hash/ time format.
